### PR TITLE
Add test for requests and limits for containers

### PR DESCRIPTION
**Pull request by sebrandon1**
Simple test to check for the existence of `requests` and `limits` on a container spec which corresponds to 4.6.11 in the new v1.4 requirements document.  Note, I don't know if we have a public link yet for the v1.4 document so I marked it as TODO.

### CODE REVIEW

File: cnf-certification-test/lifecycle/suite.go [hunk 1]
```diff
@@ -372,3 +378,23 @@ func testPodsRecreation(env *provider.TestEnvironment) { //nolint:funlen
...
 }
+
+func testPodRequestsAndLimits(env *provider.TestEnvironment) {
+	var containersMissingRequestsOrLimits []string
+	ginkgo.By("Testing container resource requests and limits")
+
+	// Loop through the containers, looking for containers that are missing requests or limits.
+	// These need to be defined in order to pass.
+	for _, cut := range env.Containers {
+		if len(cut.Data.Resources.Limits) == 0 || len(cut.Data.Resources.Requests) == 0 {
```
Comment (rdavid) [inline cnf-certification-test/lifecycle/suite.go:389]: What is cut?

I think it is more clear to have two different conditions with two different messages. By the way one container could fail both conditions.

**Review by rdavid — CHANGES_REQUESTED** [review]

File: cnf-certification-test/lifecycle/suite.go (hunk 1 above)
Comment (sebrandon1) [reply cnf-certification-test/lifecycle/suite.go:389]: This is true.  @mwlinca and I have been talking about this as well and I think there is more to add.  I'm going to make this work in progress as I fix.

**Review by rdavid — APPROVED** [review]

File: cnf-certification-test/lifecycle/suite.go [hunk 2]
```diff
@@ -372,3 +378,33 @@ func testPodsRecreation(env *provider.TestEnvironment) { //nolint:funlen
...
+
+func testPodRequestsAndLimits(env *provider.TestEnvironment) {
+	var containersMissingRequestsOrLimits []string
+	ginkgo.By("Testing container resource requests and limits")
+
+	// Loop through the containers, looking for containers that are missing requests or limits.
+	// These need to be defined in order to pass.
+	for _, cut := range env.Containers {
+		badContainer := false
+		if len(cut.Data.Resources.Limits) == 0 {
```
Comment (jmontesi) [inline cnf-certification-test/lifecycle/suite.go:390]: The requirement states that "Pods must define requests and limits values for **CPU** and **memory**" So, shouldn't this condition also check for the existence of the strings "cpu" and "memory" in the map?

File: cnf-certification-test/lifecycle/suite.go [hunk 3]
```diff
@@ -372,3 +378,33 @@ func testPodsRecreation(env *provider.TestEnvironment) { //nolint:funlen
...
+			tnf.ClaimFilePrintf("Container has been found missing resource limits: %s", cut.String())
+			badContainer = true
+		}
+
+		if len(cut.Data.Resources.Requests) == 0 {
```
Comment (jmontesi) [inline cnf-certification-test/lifecycle/suite.go:395]: Same as above.

File: cnf-certification-test/lifecycle/suite.go [hunk 4]
```diff
@@ -372,3 +378,33 @@ func testPodsRecreation(env *provider.TestEnvironment) { //nolint:funlen
...
+			badContainer = true
+		}
+
+		if badContainer {
+			containersMissingRequestsOrLimits = append(containersMissingRequestsOrLimits, cut.String())
+		}
+	}
+
+	if n := len(containersMissingRequestsOrLimits); n > 0 {
+		errMsg := fmt.Sprintf("Containers found that are missing either resource requests or limits: %d. See logs for more detail.", n)
```
Comment (jmontesi) [inline cnf-certification-test/lifecycle/suite.go:406]: If the above applies, the string should reference "requests or limits for cpu/memory".

File: cnf-certification-test/lifecycle/suite.go (hunk 2 above)
Comment (sebrandon1) [reply cnf-certification-test/lifecycle/suite.go:390]: Good call, I can get more granular instead of just checking the top level.  Will add.

Comment (sebrandon1) [reply cnf-certification-test/lifecycle/suite.go:397]: Okay I added checks to make sure that CPU and memory are set.

**Review by jmontesi — APPROVED** [review]

**Pull request merged**